### ISSUE TITLE
TabBar: Tab is closed accidentally

### ISSUE THREAD

**Issue by spoenemann** [phosphorjs/phosphor#328]
I solved #173 in theia-ide/theia#1445 by adding scroll bars to tab bars. I also added a listener to the `currentChanged` signal that scrolls to the new active tab in order to reveal it. That works great, but sometimes a tab that is partially visible is closed after clicking it.

The reason for this problem:

 * The tab is activated on `mousedown`.
 * `currentChanged` is emitted and my listener makes the tab fully visible by moving the scroll bar.
 * With some bad luck, the close icon is right below the mouse cursor after scrolling.
 * On `mouseup` the tab bar determines that the event target is the close icon, thus it closes the tab.

In summary: TabBar must not assume that the event target is the same between `mousedown` and `mouseup`, even when the mouse has not been moved in the meantime.

Possible solution: Remember the event target on `mousedown`, and on `mouseup` check whether the event target is still the same before firing `_tabCloseRequested`.
